src/main/java/ch/njol/skript/expressions/ExprDequeuedQueue.java

```diff
@@ -0,0 +1,73 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.registrations.Feature;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.lang.util.SkriptQueue;
+
+@Name("De-queue Queue (Experimental)")
+@Description("""
+	Requires the <code>using queues</code> experimental feature flag to be enabled.
+	
+	Unrolls a queue into a regular list of values, which can be stored in a list variable.
+	The order of the list will be the same as the order of the elements in the queue.
+	If a list variable is set to this, it will use numerical indices.
+	The original queue is not changed.""")
+@Examples({
+	"""
+		set {queue} to a new queue
+		add "hello" and "there" to {queue}
+		set {list::*} to dequeued {queue}"""
+})
+@Since("INSERT VERSION (experimental)")
+public class ExprDequeuedQueue extends SimpleExpression<Object> {
+
+	static {
+		Skript.registerExpression(ExprDequeuedQueue.class, Object.class, ExpressionType.SIMPLE,
+			"(de|un)queued %queue%");
```

Comment:
maybe add something like `unrolled %queue%`? i think using `dequeued` is kinda confusing